core/domain/user_services.py

```diff
@@ -1033,6 +1033,7 @@ def mark_user_for_deletion(
     """
     user_settings = get_user_settings(user_id, strict=True)
     user_settings.to_be_deleted = True
+    user_settings.deleted = True
```

Comment:
Can you explain the difference between to_be_deleted and deleted? The names are quite similar so it is a bit unclear.

Reply:
I realized that I don't need `to_be_deleted`, because it would serve the same purpose as already existing `deleted`.